rviz_common/test/interaction/mock_selection_renderer.hpp

```diff
@@ -94,7 +94,7 @@ class MockSelectionRenderer : public rviz_common::interaction::SelectionRenderer
     auto data = new uint8_t[Ogre::PixelUtil::getMemorySize(width, height, 1, Ogre::PF_A8R8G8B8)]();
     dst_box = Ogre::PixelBox(width, height, 1, Ogre::PF_A8R8G8B8, data);
 
-    auto dstptr = static_cast<uint32_t *>(dst_box.data);
+    auto dstptr = reinterpret_cast<uint32_t *>(dst_box.data);
```

Comment:
What's the purpose of this change?

Reply:
Sadly the `dst_box.data` type was changed from `uint32_t` capturing the four bytes per pixel in one int to `uchar` for each pixel. The layout of the data is the same which is why the `reinterpret_cast` works. I chose to first do it that way to show that I'm not really altering anything - the test is the same as before.

It's probably better to rework the setup to use `uint8_t` and I can do it if you want.

Reply:
I would like to see either a comment explaining that here (that we're interpreting it as four items in a 32-bit integer), or to see it refactored. I don't mind which though.

Reply:
I left a comment. Note that I figured out that the change in data structure means that some `reinterpret_cast<uint8_t *>` in the `selection_manager` and `view_picker` are now superfluous. I changed them to `static_cast` to a) help explain what we are expecting at these points and b) have something in the code that'd complain if the data structure were changed again in some future update.